lib/product-import.coffee

```diff
@@ -182,6 +185,70 @@ class ProductImport
   _generateUniqueToken: ->
     _.uniqueId "#{new Date().getTime()}"
 
+  _fetchAndResolveCustomReferences: (product) =>
+    new Promise (resolve) =>
+      if product.masterVariant
+        @_fetchAndResolveCustomReferencesByVariant(product.masterVariant).then (result) -> product.masterVariant = result
+
+      if product.variants
+        Promise.map product.variants, (variant) =>
+          @_fetchAndResolveCustomReferencesByVariant(variant)
+          .then ->
+            Promise.resolve()
+        ,{concurrency: 5}
+      resolve(product)
+
+  _fetchAndResolveCustomReferencesByVariant: (variant) =>
+    new Promise (resolve) =>
+      if variant.attributes
```

Comment:
oh now... I see. u meant absent attributes.
Updated for absent and empty attributes.